extensions/interactions/MultipleChoiceInput/directives/oppia-interactive-multiple-choice-input.directive.ts

```diff
@@ -77,8 +77,32 @@ angular.module('oppia').directive('oppiaInteractiveMultipleChoiceInput', [
           };
 
           ctrl.$onInit = function() {
-            ctrl.choices = HtmlEscaperService.escapedJsonToObj(
+            var showChoicesInShuffledOrder =
+            ($attrs.showChoicesInShuffledOrderWithValue === 'true');
+            var choicesWithValue = HtmlEscaperService.escapedJsonToObj(
               $attrs.choicesWithValue);
+            var choicesWithIndex = choicesWithValue.map(
+              function(value, originalIndex) {
+                return {originalIndex: originalIndex, value: value};
+              }
+            );
+            var shuffleChoices = function(choices) {
+              var currentIndex = choices.length;
+              var temporaryValue = null;
+              var randomIndex = null;
+
+              while (0 !== currentIndex) {
```

Comment:
Any reason not to do this as a downward-counting for loop? In general for loops are safer than while loops since they typically are guaranteed to terminate.

(I know it terminates in this case, but while loops tend to need more careful analysis for that, so I'm wondering if there's any reason not to go with the "simpler" option.)

Reply:
I have changed it into for loop. There was no particular reason for using while loop. Done.